fsharp-backend/src/ClientTypes/ClientProgramTypes.fs

```diff
@@ -0,0 +1,224 @@
+/// See ProgramTypes.fs for documentation of these types
+module ClientTypes.Program
+
+open Prelude
+open Tablecloth
+
+type id = Prelude.id
+type tlid = Prelude.tlid
+type Sign = Prelude.Sign
+
+type Position = { x : int; y : int }
+
+module FQFnName =
+  type StdlibFnName = { module_ : string; function_ : string; version : int }
+
+  type InfixStdlibFnName = { module_ : Option<string>; function_ : string }
+
+  type UserFnName = string
+
+  type PackageFnName =
+    { owner : string
+      package : string
+      module_ : string
+      function_ : string
+      version : int }
+
+  type T =
+    | User of UserFnName
+    | Stdlib of StdlibFnName
+    | Package of PackageFnName
+
+type Pattern =
+  | PVariable of id * string
+  | PConstructor of id * string * List<Pattern>
+  | PInteger of id * int64
+  | PBool of id * bool
+  | PCharacter of id * string
+  | PString of id * string
+  | PFloat of id * Sign * string * string
+  | PNull of id
+  | PBlank of id
+  | PTuple of id * Pattern * Pattern * List<Pattern>
+
+
+type SendToRail =
+  | Rail
+  | NoRail
+
+
+type Expr =
+  | EInteger of id * int64
+  | EBool of id * bool
+  | EString of id * string
+  | ECharacter of id * string
+  | EFloat of id * Sign * string * string
+  | ENull of id
+  | EBlank of id
+  | ELet of id * string * Expr * Expr
+  | EIf of id * Expr * Expr * Expr
+  | EBinOp of id * FQFnName.InfixStdlibFnName * Expr * Expr * SendToRail
+  | ELambda of id * List<id * string> * Expr
+  | EFieldAccess of id * Expr * string
+  | EVariable of id * string
+  | EFnCall of id * FQFnName.T * List<Expr> * SendToRail
+  | EPartial of id * string * Expr
+  | ERightPartial of id * string * Expr
+  | ELeftPartial of id * string * Expr
+  | EList of id * List<Expr>
+  | ETuple of id * Expr * Expr * List<Expr>
+  | ERecord of id * List<string * Expr>
+  | EPipe of id * Expr * Expr * List<Expr>
+  | EConstructor of id * string * List<Expr>
+  | EMatch of id * Expr * List<Pattern * Expr>
+  | EPipeTarget of id
+  | EFeatureFlag of id * string * Expr * Expr * Expr
+
+
+type DType =
+  | TInt
+  | TFloat
+  | TBool
+  | TNull
+  | TStr
+  | TList of DType
+  | TTuple of DType * DType * List<DType>
+  | TDict of DType
+  | TIncomplete
+  | TError
+  | THttpResponse of DType
+  | TDB of DType
+  | TDate
+  | TChar
+  | TPassword
+  | TUuid
+  | TOption of DType
+  | TErrorRail
+  | TUserType of string * int
+  | TBytes
+  | TResult of DType * DType
+  | TVariable of string
+  | TFn of List<DType> * DType
+  | TRecord of List<string * DType>
+  | TDbList of DType
+
+
+module Handler =
+  type CronInterval =
+    | EveryDay
+    | EveryWeek
+    | EveryFortnight
+    | EveryHour
+    | Every12Hours
+    | EveryMinute
+
+  // TODO This comment seems...suspicious. I wonder if this can be removed or something? @pbiggar
+  // We need to keep the IDs around until we get rid of them on the client
+  type ids = { moduleID : id; nameID : id; modifierID : id }
```

Comment:
Happy to leave this for a follow-up PR, but "We need to keep the IDs around until we get rid of them on the client" felt like we might be able to remove this from ClientTypes - for now, I'd just copied it straight from ProgramTypes.fs

Reply:
This comment is still valid I'm pretty sure.

Reply:
The comment is valid. I was wondering if `ids` could be removed, as "until we get rid of them on the client" implies that they're not _useful_ in the client.
Do we get rid of them in the client? If so, where? It seems we might not want to get rid of them in the client in any case - don't we need those IDs (at least one of them?) to reference them for updates?

Reply:
We need them while we update them via the BlankOr mechanism. When we extend fluid to handler specs, I expect we won't use IDs in handler specs anymore.